Doc/library/asyncio-task.rst

```diff
@@ -243,6 +247,11 @@ Running an asyncio Program
 Creating Tasks
 ==============
 
+**Source code:** :source:`Lib/asyncio/tasks.py`,
+:source:`Lib/asyncio/base_tasks.py`
+
+-----------------------------------
```

Comment:
I think this link is unnecessary -- too much code, most of which is unrelated and/or arcane.

Reply:
Are you referring specifically to the to link `base_tasks.py` or both links? I think we could remove `base_tasks.py`.

Reply:
I refer to both.  This documentation page is all about high-level APIs and how to use them correctly.  Bothering the reader with advanced source code isn't helpful -- those who really want to see it can easily look it up themselves.

So I'm fine with adding source links to low-level sections of the documentation, but against doing that in the high-level sections.

Sorry, I'm a bit anal about asyncio docs -- I spent 2 weeks full time rewriting them from scratch so that they are easier to follow. Overloading them with extra details doesn't serve that goal well.

Reply:
@1st1 I get it, I understand your desire to keep the docs as succinct as possible and I can see how the source code for both might be arcane in this case. So, I'll remove the links to `tasks.py` and `base_tasks.py`. 

However, I do think `runners.py` has a simple enough code to justify keeping the link, despite it being a source code link in this same page. At least from personal experience, it was really helpful to see how it works and the actual Python file location of `asyncio.run()` may not be obvious to readers.

> So I'm fine with adding source links to low-level sections of the documentation, but against doing that in the high-level sections.

My only concern with this is that the distinction between high-level and low-level API can significantly change over time. Also, the difference is not entirely black and white, as the front-page of the asycnio docs might imply (https://docs.python.org/3.9/library/asyncio.html). 

For example, consider the "Synchronization Primitives" section (https://docs.python.org/3/library/asyncio-sync.html). I personally don't find `locks.py` to be at all arcane (unlike `tasks.py` and `base_tasks.py`). Also, although it's the highest level of the API for using asynchronous locks, I don't consider it be high-level in terms of having a large degree of abstraction.

So I think that rather than having a general blanket statement of only having source links in the low-level parts, the decision should be on more of a case-by-case basis.

> Sorry, I'm a bit anal about asyncio docs

That's okay. Similar to the code for asyncio, I'd much rather we take the time to strongly consider and discuss any changes made. IMO, it's definitely worthwhile considering the number of users affected; even for minor changes.

Reply:
I removed the source links to `tasks.py` and `base_tasks.py` as requested. Leaving this open though so that you can potentially respond to my concern regarding high-level vs low-level.

Reply:
> So I think that rather than having a general blanket statement of only having source links in the low-level parts, the decision should be on more of a case-by-case basis.

Yes, and that's what we are trying to do here :)

I agree with you on your points.  In this particular case, the docs page about coroutines is all about high-level asyncio concepts; overloading it with links to tasks.py wouldn't help most readers, hence the decision not to include those links.